lexers/bash.lua

```diff
@@ -27,11 +27,20 @@ local sq_str = lexer.range("'", false, false)
 local dq_str = lexer.range('"')
 local ex_str = lexer.range('`')
 local heredoc = '<<' * P(function(input, index)
-  local _, e, _, delimiter = input:find(
-    '^%-?(["\']?)([%a_][%w_]*)%1[\n\r\f;]+', index)
-  if not delimiter then return end
-  _, e = input:find('[\n\r\f]+' .. delimiter, e)
-  return e and e + 1 or #input + 1
+  local s, e, minus, _, delimiter =
+    input:find('(-?)%s*(["\']?)([%w_]+)%2[^\n]*[\n\r\f;]+', index)
+  if s == index and delimiter then
```

Comment:
Is there a reason you removed the '^' pattern anchor and are instead checking if `s == index`? If you put the anchor back and remove the `s == index` check, does the fix still behave as expected? I ask because '^' in Lua patterns makes matching much faster in general.